xaynetwork/xayn_discovery_app
# TB-3783  fix article can be bookmarked in feed and search

### What 🕵️ 🔍

- What does the pull request solve?
The same article, in terms of url, has different ids for feed and search. 
Therefore we cannot use the article id as id when bookmarking an article, otherwise we are allowing the user to bookmark the same article twice, in the feed and in the search.
- What exactly was added or modified?
Adds the url parameter to the bookmark object
Uses the url of the article as bookmark id

NOTE: When bookmarking an article, the query parameters are removed from the url in order to keep it more simple

----------

### How to test 🥼 🔬

With a fresh install:

- Bookmark an article in the feed
- Search for the same article in the search and check that it's marked as bookmarked
- If you unbookmark the article it should unbookmarked for both screen, feed and search

Install on top of a previous build, where you have bookmarked the same article twice (please check that the provider is the same) and check that one of the two bookmarks has been removed. This scenario is shown with the before and after videos.

Since this is a quite big refactor, please check that the entire bookmark and collection flow still works as expected, like creating, deleting and moving bookmarks

----------

### Screenshots 📸 📱

| Before | After  |
| ------ | ------ |
| <video width="280" src="https://user-images.githubusercontent.com/74236996/173661674-38eade38-687b-4c1a-9b0a-ef8cd8492995.mov"> | <video width="280" src="https://user-images.githubusercontent.com/74236996/173661887-1d2d2024-e3a2-4f07-9714-248ae0264393.mov"> |

----------

### References 📝 🔗

- [JIRA](https://xainag.atlassian.net/browse/TB-3783)

----------
